fitbenchmarking/results_processing/tests/test_problem_summary_page.py

```diff
@@ -151,13 +151,12 @@ def setUp(self):
         """
         Setup for create function tests
         """
-        self.results_dir = TemporaryDirectory()
-        self.supp_dir = os.path.join(self.results_dir.name, 'support_pages')
-        self.fig_dir = os.path.join(self.supp_dir, 'figures')
+        with TemporaryDirectory() as directory:
+            directory_name = directory
+            self.supp_dir = os.path.join(directory_name, 'support_pages')
+            self.fig_dir = os.path.join(self.supp_dir, 'figures')
         os.makedirs(self.fig_dir)
```

Comment:
I think this will need to be indented to allow tempdir to cleanup properly. As it is, I **think** it will do the cleanup before makdirs which will then create permanent versions.

Reply:
I think these directories need to stick around as they're required for the tests. But I've added a `tearDown` method to remove them once the tests are finished.

Reply:
Sorry I thought this was in a specific test not a setup. I can't see a tearDown, did it get missed in the commit?
I think the way it was previously, handled teardown automatically as TeporaryDirectory deletes the directory when the object is destroyed.